go/stellar/stellar.go

```diff
@@ -697,3 +700,45 @@ func identifyRecipient(m libkb.MetaContext, assertion string) (keybase1.TLFIdent
 
 	return frep, nil
 }
+
+func FormatCurrency(ctx context.Context, g *libkb.GlobalContext, amount string, code stellar1.OutsideCurrencyCode) (string, error) {
+	conf, err := g.GetStellar().GetServerDefinitions(ctx)
+	if err != nil {
+		return "", err
+	}
+	currency, ok := conf.Currencies[code]
+	if !ok {
+		return "", fmt.Errorf("Could not find currency %q", code)
+	}
+
+	amountFmt, err := FormatAmount(amount, true)
+	if err != nil {
+		return "", err
+	}
+
+	if currency.Symbol.Postfix {
+		return fmt.Sprintf("%s %s", amountFmt, currency.Symbol.Symbol), nil
+	}
+
+	return fmt.Sprintf("%s%s", currency.Symbol.Symbol, amountFmt), nil
+}
+
+func FormatAmount(amount string, precisionTwo bool) (string, error) {
+	if amount == "" {
+		amount = "0"
```

Comment:
This strikes me as the last place an important error would be caught. Can this error out? Any reason to expect its input to be `""`?